### PR TITLE
Change `ADO.NET` to more specific namespaces

### PR DESCRIPTION
**Pull request by bouwkast**
### What does this PR do? What is the motivation?

Breaks up `ADO.NET` into `System.Data.SqlClient` and `Microsoft.Data.SqlClient`
We don't fully support `ADO.NET` for DBM, but we _do_ support it via these two namespaces.

### Merge instructions

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/database_monitoring/connect_dbm_and_apm.md [hunk 1]
```diff
@@ -448,11 +449,12 @@ View historical performance of similar queries to those executed in your trace,
...
-[24]: https://learn.microsoft.com/en-us/dotnet/framework/data/adonet/ado-net-overview
+[24]: https://learn.microsoft.com/sql/connect/ado-net/microsoft-ado-net-sql-server
 [25]: https://learn.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlcommand.commandtype?view=dotnet-plat-ext-7.0#remarks:~:text=[…]%20should%20set
 [26]: https://app.datadoghq.com/services
 [27]: https://pypi.org/project/asyncpg/
 [28]: https://pypi.org/project/aiomysql/
 [29]: https://pypi.org/project/mysql-connector-python/
 [30]: https://pypi.org/project/mysqlclient/
 [31]: https://github.com/PyMySQL/PyMySQL
+[32]: https://learn.microsoft.com/sql/connect/ado-net/introduction-microsoft-data-sqlclient-namespace?view=sql-server-ver16
```
Comment (vandonr) [inline content/en/database_monitoring/connect_dbm_and_apm.md:460]:
```suggestion
[32]: https://learn.microsoft.com/sql/connect/ado-net/introduction-microsoft-data-sqlclient-namespace
```

Comment (bouwkast) [reply content/en/database_monitoring/connect_dbm_and_apm.md:460]: Thanks!

**Review by vandonr — APPROVED** [review]
LGTM

**Review by lucaspimentel — APPROVED** [review]

File: content/en/database_monitoring/connect_dbm_and_apm.md [hunk 2]
```diff
@@ -72,7 +73,7 @@ APM tracer integrations support a *Propagation Mode*, which controls the amount
 
 \* [CommandType.StoredProcedure][25] not supported
 
-\*\* Full mode SQL Server/Java:
+\*\* Full mode SQL Server for Java/.NET:
```
Comment (drichards-87) [inline content/en/database_monitoring/connect_dbm_and_apm.md:76]: Can we indent the bulleted list that follows this sentence?

**Review by drichards-87 — CHANGES_REQUESTED** [review]

File: content/en/database_monitoring/connect_dbm_and_apm.md (hunk 2 above)
Comment (bouwkast) [reply content/en/database_monitoring/connect_dbm_and_apm.md:76]: Yep indented it thanks!

**Review by drichards-87 — APPROVED** [review]

**Pull request merged**